src/Microsoft.ML.FastTree/TreeEnsembleFeaturizationEstimator.cs

```diff
@@ -0,0 +1,464 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Linq;
+using Microsoft.ML.Data;
+using Microsoft.ML.Runtime;
+
+namespace Microsoft.ML.Trainers.FastTree
+{
+    /// <summary>
+    /// This class encapsulates the common behavior of all tree-based featurizers such as <see cref="FastTreeBinaryFeaturizationEstimator"/>,
+    /// <see cref="FastForestBinaryFeaturizationEstimator"/>, <see cref="FastTreeRegressionFeaturizationEstimator"/>,
+    /// <see cref="FastForestRegressionFeaturizationEstimator"/>, and <see cref="PretrainedTreeFeaturizationEstimator"/>.
+    /// All tree-based featurizers share the same output schema computed by <see cref="GetOutputSchema(SchemaShape)"/>. All tree-based featurizers
+    /// requires an input feature column name and a suffix for all output columns. The <see cref="ITransformer"/> returned by <see cref="Fit(IDataView)"/>
+    /// produces three columns: (1) the prediction values of all trees, (2) the IDs of leaves the input feature vector falling into, and (3)
+    /// the binary vector which encodes the paths to those destination leaves.
+    /// </summary>
+    public abstract class TreeEnsembleFeaturizationEstimatorBase : IEstimator<TreeEnsembleFeaturizationTransformer>
+    {
+        /// <summary>
+        /// The common options of tree-based featurizations such as <see cref="FastTreeBinaryFeaturizationEstimator"/>, <see cref="FastForestBinaryFeaturizationEstimator"/>,
+        /// <see cref="FastTreeRegressionFeaturizationEstimator"/>, <see cref="FastForestRegressionFeaturizationEstimator"/>, and <see cref="PretrainedTreeFeaturizationEstimator"/>.
+        /// </summary>
+        public class CommonOptions
```

Comment:
Why is this named `CommonOptions`? I think we normally name a base class `OptionsBase`.

For example: [https://github.com/dotnet/machinelearning/blob/a1b5eaabedda87554328682c4dfd3aa311463002/src/Microsoft.ML.StandardTrainers/Standard/SdcaBinary.cs#L157](https://github.com/dotnet/machinelearning/blob/a1b5eaabedda87554328682c4dfd3aa311463002/src/Microsoft.ML.StandardTrainers/Standard/SdcaBinary.cs#L157)

Reply:
No problem. I am renaming it.

---
In reply to: [291369495](https://github.com/dotnet/machinelearning/pull/3812#discussion_r291369495) [](ancestors = 291369495)